pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsReaderImpl.java

```diff
@@ -83,8 +83,13 @@ public MultiTopicsReaderImpl(PulsarClientImpl client, ReaderConfigurationData<T>
 
                 @Override
                 public void received(Consumer<T> consumer, Message<T> msg) {
+                    final MessageId messageId = msg.getMessageId();
+                    consumer.acknowledgeCumulativeAsync(messageId).exceptionally(ex -> {
+                        log.error("[{}][{}] auto acknowledge message {} cumulative fail.", getTopic(),
+                                getMultiTopicsConsumer().getSubscription(), msg.getMessageId(), ex);
+                        return null;
+                    });
```

Comment:
Is it better to execute the acknowledgment after triggering the `readerListener`? You already have the message ID, and it is immutable, right? So that we will not change any behavior here.

Reply:
Ah, a good idea to keep behaviour.

Reply:
Fixed. PTAL.